apps/meteor/app/livechat/server/lib/LivechatTyped.ts

```diff
@@ -459,6 +462,38 @@ class LivechatClass {
 			extraData,
 		});
 
+		if (isSingleContactEnabled()) {
+			let { contactId } = visitor;
+			if (!contactId) {
+				const visitorContact = await LivechatVisitors.findOne<Pick<ILivechatVisitor, 'contactId'>>(visitor._id, {
+					projection: { contactId: 1 },
+				});
+				contactId = visitorContact?.contactId;
+			}
+
+			const contact = await LivechatContacts.findOneById<Pick<ILivechatContact, '_id' | 'channels'>>(contactId as string, {
```

Comment:
A visitor may not have a contactId (specially for visitors created before this call) and this will make it pass `undefined` to the findOneById, which iirc will return null, but at the same time is a db call we can save by validating contactId being an string before calling. wdyt?

Reply:
Good catch! I added a method to create the contact if the visitor doesn't have one. We'll need it anyway, so it's an excellent way to ensure that we migrate old visitors by demand.